packages/lesswrong/server/resolvers/ultraFeedResolver.ts

```diff
@@ -0,0 +1,571 @@
+         console.warn("UltraFeedResolver: Fetched spotlights but 'spotlights' source is not defined in sourceWeights.");
+      }
+
+      postThreadsItems.forEach(postItem => {
+        const itemSources = postItem.postMetaInfo?.sources;
+        if (Array.isArray(itemSources)) {
+          itemSources.forEach(source => {
+            const sourceType = source as FeedItemSourceType;
+            if (sources[sourceType]) {
+              sources[sourceType].items.push(postItem);
+            }
+          });
+        }
+      });
+
+      commentThreadsItems.forEach(commentThread => {
+        let foundSources = false;
+        // Find the first comment that *has* source info
+        for (const comment of commentThread.comments) {
+          const itemSources = comment?.metaInfo?.sources;
+          if (Array.isArray(itemSources) && itemSources.length > 0) {
+            itemSources.forEach(source => {
+              const sourceType = source as FeedItemSourceType;
+              if (sources[sourceType]) {
+                sources[sourceType].items.push(commentThread);
+                foundSources = true;
+              }
+            });
+            if (foundSources) {
+              break;
+            }
+          }
+        }
+      });
+
+      const populatedSources = Object.entries(sources).reduce((acc, [key, value]) => {
+        if (value.items.length > 0) {
+          acc[key as FeedItemSourceType] = value;
+        }
+        return acc;
+      }, {} as Record<FeedItemSourceType, WeightedSource>);
+
+      const sampledItems = weightedSample(populatedSources, limit);
+      
+      const spotlightIdsToLoad: string[] = [];
+      const commentIdsToLoad = new Set<string>();
+
+      sampledItems.forEach(item => {
+        if (item.renderAsType === "feedSpotlight") {
+          spotlightIdsToLoad.push(item.feedSpotlight.spotlightId);
+        } else if (item.renderAsType === "feedCommentThread") {
+          item.feedCommentThread.comments?.forEach(comment => {
+            if (comment.commentId) {
+              commentIdsToLoad.add(comment.commentId);
+            }
+          });
+        }
+      });
+      
+      const uniqueCommentIds = Array.from(commentIdsToLoad);
+      
+      const [spotlightsResults, commentsResults] = await Promise.all([
+        context.loaders.Spotlights.loadMany(spotlightIdsToLoad),
+        context.loaders.Comments.loadMany(uniqueCommentIds)
+      ]);
+      
+      const spotlightsById = new Map<string, DbSpotlight>();
+      spotlightsResults.forEach(result => {
+        if (result instanceof Error) {
+          // eslint-disable-next-line no-console
+          console.error("Error loading spotlight:", result);
+        } else if (result?._id) {
+          spotlightsById.set(result._id, result);
+        }
+      });
+
+      const commentsById = new Map<string, DbComment>();
+      commentsResults.forEach(result => {
+        if (result instanceof Error) {
```

Comment:
Just use `loadByIds` instead of the loader's `loadMany` to avoid needing to deal with errors like ths.

Reply:
Then you can just do this in a single line:
```
const commentsById = keyBy(commentResults, '_id');
```